### PR TITLE
Fetch the latest micro version of python from conda in cross version tests to avoid `ResolvePackageNotFound` error

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/cross-version-tests.yml [hunk 1]
```diff
@@ -78,7 +78,8 @@ jobs:
           else
             python_version=3.6
           fi
-          echo "::set-output name=python-version::$python_version"
+          latest_micro_version=$($CONDA/bin/conda search python | grep $python_version | tail -1 | tr -s ' ' | cut -d' ' -f2)
+          echo "::set-output name=python-version::$latest_micro_version"
```
Comment (WeichenXu123) [inline .github/workflows/cross-version-tests.yml:82]: should this be:
```
echo "::set-output name=python-version::${python_version}.$latest_micro_version"
```

Comment (harupy) [reply .github/workflows/cross-version-tests.yml:82]: Offline discussion: we agreed on the current code and variable name.

File: .github/workflows/cross-version-tests.yml [hunk 2]
```diff
@@ -78,7 +78,8 @@ jobs:
           else
             python_version=3.6
           fi
-          echo "::set-output name=python-version::$python_version"
+          latest_micro_version=$($CONDA/bin/conda search python | grep -F $python_version | tail -1 | tr -s ' ' | cut -d' ' -f2)
```
Comment (harupy) [inline .github/workflows/cross-version-tests.yml:81]: `conda search python` outputs:

```
python                        3.6.10      h359304d_0  pkgs/main           
python                        3.6.10      hc70fcce_1  pkgs/main           
python                        3.6.10      hf48f09d_2  pkgs/main           
python                        3.6.10      hfe9666f_1  pkgs/main           
python                        3.6.12      h26836e1_2  pkgs/main           
python                        3.6.13      h88f2d9e_0  pkgs/main           
python                         3.7.0      hc167b69_0  pkgs/main           
python                         3.7.1      haf84260_3  pkgs/main   
```

`grep -F $python_version`: Captures specified minor python versions
`tail -1`: Fetches the last row
`tr -s ' '`: Shrinks spaces
`cut -d' ' -f2`: Splits the row with a space and get the second element.

**Comment by harupy** [comment]
@WeichenXu123 Confirmed the python 3.6.13 is used and the tests works now:

https://github.com/mlflow/mlflow/pull/4559/checks?check_run_id=3065363318#step:5:3

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**